source/frontend/src/features/properties/map/project/add/ProductSubForm.tsx

```diff
@@ -0,0 +1,75 @@
+import { FastDatePicker, Input, TextArea } from 'components/common/form';
+import { SectionField } from 'features/mapSideBar/tabs/SectionField';
+import { FormikProps } from 'formik';
+import React from 'react';
+import { Col, Row } from 'react-bootstrap';
+import styled from 'styled-components';
+import { withNameSpace } from 'utils/formUtils';
+
+import { ProjectForm } from './models';
+
+export interface IProductSubFormProps {
+  index: number;
+  nameSpace: string;
+  formikProps: FormikProps<ProjectForm>;
+}
+
+export const ProductSubForm: React.FunctionComponent<IProductSubFormProps> = ({
+  index,
+  nameSpace,
+  formikProps,
+}) => {
+  const costEstimate = formikProps.values.products[index].costEstimate;
```

Comment:
might want to use withNameSpace here as well?

Reply:
imo this is cleaner and type-checkeable. But no prob switching to the formik way if you believe strongly on it

Reply:
Not that strongly, just noticing it is being used below but not here. Was wondering if there was something I was missing.